### PR TITLE
Add iPadOS 13.4 Pointer Interaction for Panel Resizing

### LINKED ISSUE #32: Pointer Interaction
**Describe the bug**
When using a pointing devices on iOS 13.4 the pointer doesn't adopt to highlight the resize handle

**To Reproduce**
Move with the pointer over the resize handle

**Expected behavior**
Similar to Maps app the mouse pointer changes to highlight the bar

### PR DESCRIPTION
**Pull request by myell0w**
Fixes #32 

Improves resizing of the popover with an iPadOS 13.4 Pointer

TODO:
 - [ ] Add New Tag
 - [ ] Update Carthage Dependency in MindNode

### CODE REVIEW

File: Aiolos/Aiolos/Sources/PanelGestures.swift [hunk 1]
```diff
@@ -374,8 +374,6 @@ private extension PanelGestures {
         }
 
         private func handlePanChanged(_ pan: PanGestureRecognizer) {
-            guard let parentView = self.panel.parent?.view else { return }
```
Comment (myell0w) [inline Aiolos/Aiolos/Sources/PanelGestures.swift]: wasn't used anywhere

File: Aiolos/Aiolos/Sources/ResizeHandle.swift [hunk 2]
```diff
@@ -14,9 +14,13 @@ public final class ResizeHandle: UIView {
...
     public struct Constants {
         public static let height: CGFloat = 20.0
+
+        fileprivate static let handleHeight: CGFloat = 5.0
+        fileprivate static let inactiveHandleWidth: CGFloat = 38.0
+        fileprivate static let activeHandleWidth: CGFloat = 44.0
     }
 
-    private lazy var resizeHandle: CAShapeLayer = self.makeResizeHandle()
+    private lazy var resizeHandle: UIView = self.makeResizeHandle()
```
Comment (myell0w) [inline Aiolos/Aiolos/Sources/ResizeHandle.swift]: replaced the CALayer with an instance of UIView, to simplify the pointer delegate

File: Aiolos/Aiolos/Sources/ResizeHandle.swift [hunk 3]
```diff
@@ -49,22 +54,34 @@ public final class ResizeHandle: UIView {
...
+        }
     }
 
     public required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
+    // MARK: - UIView
+
+    override public func layoutSubviews() {
```
Comment (myell0w) [inline Aiolos/Aiolos/Sources/ResizeHandle.swift]: was moved upwards from an extension below

File: Aiolos/Aiolos/Sources/ResizeHandle.swift [hunk 4]
```diff
@@ -77,65 +94,66 @@ extension ResizeHandle {
...
+// MARK: - UIPointerInteractionDelegate
 
-extension ResizeHandle {
+@available(iOS 13.4, *)
+extension ResizeHandle: UIPointerInteractionDelegate {
 
-    override public func layoutSubviews() {
-        super.layoutSubviews()
+    public func pointerInteraction(_ interaction: UIPointerInteraction, regionFor request: UIPointerRegionRequest, defaultRegion: UIPointerRegion) -> UIPointerRegion? {
+        return UIPointerRegion(rect: self.resizeHandle.frame.insetBy(dx: -24.0, dy: -12.0))
```
Comment (myell0w) [inline Aiolos/Aiolos/Sources/ResizeHandle.swift]: not sure which values to use here, used this to extend the area where the pointer snaps

File: Aiolos/Aiolos/Sources/ResizeHandle.swift [hunk 5]
```diff
@@ -77,65 +94,66 @@ extension ResizeHandle {
...
+      }
+
+    public func pointerInteraction(_ interaction: UIPointerInteraction, styleFor region: UIPointerRegion) -> UIPointerStyle? {
+        return UIPointerStyle(effect: .automatic(UITargetedPreview(view: self.resizeHandle)))
```
Comment (myell0w) [inline Aiolos/Aiolos/Sources/ResizeHandle.swift]: this is not equal to the pointer design in the Maps App, but the best I was able to do in a not too-long of time, without having to deal with custom UIBezierPaths for the shape

File: Aiolos/Aiolos/Sources/ResizeHandle.swift [hunk 6]
```diff
@@ -77,65 +94,66 @@ extension ResizeHandle {
...
+    }
 
-        self.resizeHandle.frame = self.bounds
-        self.updateResizeHandlePath()
+    public func pointerInteraction(_ interaction: UIPointerInteraction, willEnter region: UIPointerRegion, animator: UIPointerInteractionAnimating) {
+        self.isResizing = true
```
Comment (myell0w) [inline Aiolos/Aiolos/Sources/ResizeHandle.swift]: Imo it's a nice touch to reuse our existing bigger handle to indicate pointer interaction

File: Aiolos/Aiolos/Sources/ResizeHandle.swift [hunk 7]
```diff
@@ -77,65 +94,66 @@ extension ResizeHandle {
...
-    }
-
-    func addAnimation(to layer: CAShapeLayer) {
-        let animationKey = "pathAnimation"
-        let animation = CABasicAnimation(keyPath: #keyPath(CAShapeLayer.path))
-        animation.duration = 0.3
-        animation.timingFunction = CAMediaTimingFunction(name: CAMediaTimingFunctionName.easeInEaseOut)
-
-        layer.removeAnimation(forKey: animationKey)
-        layer.add(animation, forKey: animationKey)
```
Comment (tomaskraina) [inline Aiolos/Aiolos/Sources/ResizeHandle.swift:135]: I like how the code can be simplified by not using `CAShapeLayer` 👍

**Pull request merged**